scripts/workspace-conversion/workspace-to-dataset.py

```diff
@@ -0,0 +1,296 @@
+            ret.append("TerraCoreValueSets:Microbiome")
+        if modality == "Metagenomic":
+            ret.append("TerraCoreValueSets:Metagenomic")
+        if modality == "Proteomic":
+            ret.append("TerraCoreValueSets:Proteomic")
+        if modality == "Transcriptomic":
+            ret.append("TerraCoreValueSets:Transcriptomic")
+        if modality == "SpatialTranscriptomics":
+            ret.append("TerraCoreValueSets:SpatialTranscriptomics")
+        if modality == "Trascriptomic_Targeted":
+            ret.append("TerraCoreValueSets:Transcriptomic_Targeted")
+        if modality == "Trascriptomic_NonTargeted":
+            ret.append("TerraCoreValueSets:Transcriptomic_NonTargeted")
+        if modality == "Trascriptomic_NonTargeted_RnaSeq":
+            ret.append("TerraCoreValueSets:Transcriptomic_NoneTargeted_RnaSeq")
+        if modality == "Trascriptomic_NonTargeted_MicroRnaCounts":
+            ret.append("TerraCoreValueSets:Transcriptomic_NonTargeted_MicroRnaCounts")
+        if modality == "Electrocardiogram":
+            ret.append("TerraCoreValueSets:Electrocardiogram")
+    return ret
+
+
+def generateCatalogMetadata(workspace):
+    print("Generating workspace metadata")
+
+    wsAttributes = workspace["workspace"]["attributes"]
+
+    # Set empty up the major objects
+    metadata = {}
+    metadata["samples"] = {}
+    metadata["samples"]["disease"] = list(
+        filter(
+            None,
+            [
+                wsAttributes.get("library:diseaseOntologyLabel", None),
+                wsAttributes.get("library:indication", None),
+                wsAttributes.get("library:primaryDiseaseSite", None),
+                wsAttributes.get("library:studyDesign", None),
+                wsAttributes.get("library:cellType", None),
+            ],
+        )
+    )
+    metadata["counts"] = {}
+    metadata["counts"]["donors"] = wsAttributes.get("library:numSubjects", 0)
+    metadata["dct:dataCategory"] = wsAttributes.get("library:dataCategory", {}).get(
+        "items", None
+    )
+    metadata["TerraDCAT_ap:hasDataUsePermission"] = list(
+        filter(
+            None,
+            [
+                mapDatasetReleasePolicy(
+                    wsAttributes.get("library:dataUseRestriction", "No restrictions")
+                )
+            ],
+        )
+    )
+    metadata["dct:title"] = wsAttributes.get("library:datasetName", None)
+    metadata["dct:version"] = wsAttributes.get("library:datasetVersion", None)
+    metadata["dct:description"] = wsAttributes.get("library:datasetDescription", None)
+    metadata["TerraDCAT_ap:hasOwner"] = wsAttributes.get("library:datasetOwner", None)
+    metadata["TerraDCAT_ap:hasDataCollection"] = []
+    if "library:datasetOwner" in wsAttributes:
+        metadata["TerraDCAT_ap:hasDataCollection"].append({})
+        metadata["TerraDCAT_ap:hasDataCollection"][0]["dct:identifier"] = wsAttributes[
+            "library:datasetOwner"
+        ]
```

Comment:
Can this dictionary be created as an argument to append? Something like
```suggestion
        metadata["TerraDCAT_ap:hasDataCollection"].append({"dct:identifier": wsAttributes["library:datasetOwner"]})
```